pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -20,7 +20,10 @@
 # TODO enable shared libs for cross-compiling
 , enableSharedExecutables ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
 , enableSharedLibraries ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
-, enableSplitObjs ? !stdenv.isDarwin # http://hackage.haskell.org/trac/ghc/ticket/4013
+# For GHC 8.0.2 we use split-sections instead
+# Unsupported for Darwin/Clang, see http://hackage.haskell.org/trac/ghc/ticket/4013
+, enableSplitObjs ? (!stdenv.isDarwin && (stdenv.lib.versionAtLeast "8.0.1" ghc.version))
+, enableSplitSections ? (!stdenv.isDarwin && (stdenv.lib.versionOlder "8.0.1" ghc.version))
```

Comment:
@peti sounds good. I think we should rename the option though, since objsplit would then imply section split for GHC 8.0.2 and we're overloading the term. Can we error out if `enableSplitObjs` is set and use general term such as `enableDeadCodeElimination`?